core/src/main/java/io/questdb/cairo/sql/SqlExecutionCircuitBreakerWrapper.java

```diff
@@ -0,0 +1,139 @@
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ ******************************************************************************/
+
+package io.questdb.cairo.sql;
+
+import io.questdb.std.MemoryTag;
+import io.questdb.std.Misc;
+import org.jetbrains.annotations.Nullable;
+import org.jetbrains.annotations.TestOnly;
+
+import java.io.Closeable;
+import java.io.IOException;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+// does not have to be thread-safe, because worker threads own the jobs/contexts
+// which hold SqlExecutionCircuitBreakerWrapper objects
+public class SqlExecutionCircuitBreakerWrapper implements SqlExecutionCircuitBreaker, Closeable {
+    private SqlExecutionCircuitBreaker delegate;
+    private NetworkSqlExecutionCircuitBreaker networkSqlExecutionCircuitBreaker;
+
+    @Override
+    public void cancel() {
+        delegate.cancel();
+    }
+
+    @Override
+    public boolean checkIfTripped(long millis, long fd) {
+        return delegate.checkIfTripped(millis, fd);
+    }
+
+    @Override
+    public boolean checkIfTripped() {
+        return delegate.checkIfTripped();
+    }
+
+    @Override
+    public void close() throws IOException {
+        delegate = Misc.freeIfCloseable(networkSqlExecutionCircuitBreaker);
+    }
+
+    @Override
+    public @Nullable SqlExecutionCircuitBreakerConfiguration getConfiguration() {
+        return delegate.getConfiguration();
+    }
+
+    @TestOnly
+    public SqlExecutionCircuitBreaker getDelegate() {
+        return delegate;
+    }
+
+    @Override
+    public long getFd() {
+        return delegate.getFd();
+    }
+
+    @Override
+    public int getState() {
+        return delegate.getState();
+    }
+
+    @Override
+    public int getState(long millis, long fd) {
+        return delegate.getState(millis, fd);
+    }
+
+    public void init(SqlExecutionCircuitBreakerWrapper wrapper) {
+        init(wrapper.delegate);
+    }
+
+    public void init(SqlExecutionCircuitBreaker executionContextCircuitBreaker) {
+        final SqlExecutionCircuitBreakerConfiguration sqlExecutionCircuitBreakerConfiguration = executionContextCircuitBreaker.getConfiguration();
+        if (sqlExecutionCircuitBreakerConfiguration != null) {
+            if (networkSqlExecutionCircuitBreaker == null) {
+                networkSqlExecutionCircuitBreaker = new NetworkSqlExecutionCircuitBreaker(sqlExecutionCircuitBreakerConfiguration, MemoryTag.NATIVE_CB2);
```

Comment:
This used to be `MemoryTag.NATIVE_CB1` in `PageFrameReduceJob`. If we use `NATIVE_CB2`, the `NATIVE_CB1` tag becomes unused and we can remove it.

Reply:
This replaces the code which was in `PageFrameSequence.initRecord()`, and there we used `NATIVE_CB2`.
`NATIVE_CB1` was used in the constructor of `PageFrameReduceJob`, that code has been removed.
You are right that `NATIVE_CB1` is not used anymore.
I will remove it.

Reply:
Actually now that I think about it more, this method is used from `PageFrameReduceJob` too, so basically the two places are merged into one. This replaces both, `NATIVE_CB1` and `NATIVE_CB2`.
Bottom line is that we can remove `NATIVE_CB1`. :)

Reply:
done

Reply:
kept the names to stay backwards compatible. :)
since this PR has been merged, can change it in another one.

Reply:
No prob at all!